config/main.py

```diff
@@ -49,9 +63,137 @@ def log_error(msg):
     syslog.closelog()
 
 #
-# Helper functions
+# Breakout Mode Helper functions
 #
 
+# Read given JSON file
+def readJsonFile(fileName):
+    try:
+        with open(fileName) as f:
+            result = json.load(f)
+    except Exception as e:
+        raise Exception(str(e))
+    return result
+
+
+def _get_child_interface_speed_dict(interface_name, breakout_mode):
+    """
+    Returns list of interfaces needed to be deleted or added
+    to change from  running breakout mode to target breakout mode
+    """
+
+    global SYMMETRIC_SPLIT_BREAKOUT_PATTERN, ASYMMETRIC_SPLIT_BREAKOUT_PATTERN
+    parent_interface_index = re.search("Ethernet(\d+)", interface_name)
+    if parent_interface_index is not None:
+        index = parent_interface_index.group(1)
+
+    """
+    Regex for  SYMMETRIC breakout_modes
+        Syntax  : number x speed1 [speed2, speed3]
+                : means the parent port lanes are split equally to number of ports denoted by the number.
+                And the speed for each breakout port is default to speed1, but is changeable to speed2, speed3 etc.
+        Example : 1x100G[40G], 2x50G, 4x25G[10G]
+    """
+    SYMMETRIC_SPLIT_BREAKOUT_PATTERN = "(^\d{1})x(\d{2,3}G)(\[\d{2,3}G\])?"
+
+    """
+    Regex for ASYMMETRIC breakout_modes
```

Comment:
Can you check whether this Regex logic could be shared with portconfig.py in sonic-config-engine? If so, we should make it as a lib.